sagemaker-mlflow/sagemaker_pipelines_mlflow.ipynb

```diff
@@ -0,0 +1,1288 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "32cbb17e-1967-45c0-a564-385fe2136dc0",
+   "metadata": {},
+   "source": [
+    "# SageMaker Pipelines with MLflow"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a0908eda",
+   "metadata": {},
+   "source": [
+    "This notebook's CI test result for us-west-2 is as follows. CI test results in other regions can be found at the end of the notebook.\n",
+    "\n",
+    "![This us-west-2 badge failed to load. Check your device's internet connectivity, otherwise the service is currently unavailable](https://prod.us-west-2.tcx-beacon.docs.aws.dev/sagemaker-nb/us-west-2/sagemaker-mlflow|sagemaker_pipelines_mlflow.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "58a10420-4496-4d45-ba93-c261c3ca2aa8",
+   "metadata": {},
+   "source": [
+    "## Setup environment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cfb5edce-c55d-4cdf-afb0-ed20a31dcba4",
+   "metadata": {},
+   "source": [
+    "Import necessary libraries"
```

Comment:
I think some of these are failing because there is no install step for dependencies

Reply:
```
* sagemaker_deployment_mlflow.ipynb ****************************************************************
*
*    job name: papermill-sagemaker-deployment-mlflow-2024-06-18-23-36-37
*
*      kernel: sagemaker-distribution-1.6-cpu
*
*    instance: ml.m5.xlarge
*
*      status: Failed     
*
failure_reason: 
---------------------------------------------------------------------------
Exception encountered at "In [3]":
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[3], line 7
      5 from sagemaker.serve import ModelBuilder
      6 from sagemaker.serve.mode.function_pointers import Mode
----> 7 import mlflow
      8 from mlflow import MlflowClient
      9 import boto3

ModuleNotFoundError: No module named 'mlflow'

*
*

---------------------------------------------------------------------------
Exception encountered at "In [3]":
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[3], line 7
      5 from sagemaker.serve import ModelBuilder
      6 from sagemaker.serve.mode.function_pointers import Mode
----> 7 import mlflow
      8 from mlflow import MlflowClient
      9 import boto3

ModuleNotFoundError: No module named 'mlflow'

```

Reply:
```
* sagemaker_hpo_mlflow.ipynb ***********************************************************************
*
*    job name: papermill-sagemaker-hpo-mlflow-2024-06-18-23-36-39
*
*      kernel: sagemaker-distribution-1.6-cpu
*
*    instance: ml.t3.medium
*
*      status: Failed     
*
failure_reason: 
---------------------------------------------------------------------------
Exception encountered at "In [3]":
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[3], line 6
      3 from torchvision import transforms
      4 from torchvision.datasets import MNIST
----> 6 import mlflow
      7 import sagemaker
      8 from sagemaker import get_execution_role

ModuleNotFoundError: No module named 'mlflow'

*
*

---------------------------------------------------------------------------
Exception encountered at "In [3]":
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[3], line 6
      3 from torchvision import transforms
      4 from torchvision.datasets import MNIST
----> 6 import mlflow
      7 import sagemaker
      8 from sagemaker import get_execution_role

ModuleNotFoundError: No module named 'mlflow'

```